src/main/kotlin/no/nav/familie/ef/sak/vedtak/VedtakController.kt

```diff
@@ -125,26 +125,31 @@ class VedtakController(private val stegService: StegService,
 
     @PostMapping("/gjeldendeIverksatteBehandlingerMedInntekt")
     @ProtectedWithClaims(issuer = "azuread", claimMap = ["roles=access_as_application"]) //Familie-ef-personhendelse bruker denne
-    fun hentPersonerMedAktivStonadOgForventetInntekt(@RequestBody personIdenter: List<String>): Ressurs<Map<String, Int?>> {
+    fun hentPersonerMedAktivStonadOgForventetInntekt(@RequestBody
+                                                     personIdenter: List<String>): Ressurs<List<ForventetInntektForPersonIdent>> {
         logger.info("hentPersonerMedAktivStonadOgForventetInntekt start")
         val personIdentToBehandlingIds = behandlingRepository.finnSisteIverksatteBehandlingerForPersonIdenter(personIdenter).toMap()
         logger.info("hentPersonerMedAktivStonadOgForventetInntekt hentet behandlinger")
-        val identToForventetInntektMap = mutableMapOf<String, Int?>()
 
-        val behandlingIdToForventetInntektMap =
+        val personIdentMedForventetInntektList = mutableListOf<PersonIdentMedForventetInntekt>()
+        val forventetInntektForBehandling =
                 vedtakService.hentForventetInntektForBehandlingIds(personIdentToBehandlingIds.values)
 
         for (personIdent in personIdentToBehandlingIds.keys) {
             val behandlingId = personIdentToBehandlingIds[personIdent]
-            if (behandlingIdToForventetInntektMap[behandlingId] == null) {
+            if (forventetInntektForBehandling.firstOrNull { it.behandlingId == behandlingId } == null) {
```

Comment:
Tenker man burde kunnet trukket ut `forventetInntektForBehandling.firstOrNull { it.behandlingId == behandlingId }` til en egen variabel her, som då kan brukes på rad 144 i stedet for å gjøre samme sak på nytt på 143